### PR TITLE
powerdns recursor plugin

### PR DESCRIPTION
**Pull request by lwhitworth**
Python plugin for powerdns recursor based off of the powerdns one that already exists

### CODE REVIEW

**Comment by l2isbad** [comment]
Thanks for contributing @lwhitworth 

Please take a look at [New Module PR Checklist](https://github.com/firehol/netdata/wiki/New-Module-PR-Checklist)

**Comment by lwhitworth** [comment]
Updated in line with guidelines.  Few minor edits for clarity, one extra metric and one mistake (apologies for that one!).

**Comment by l2isbad** [comment]
Sign the [CLA](https://github.com/firehol/netdata/blob/master/CONTRIBUTORS.md) please

File: conf.d/python.d/powerdns_recursor.conf [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+# job_name:
+#     name: myname            # the JOB's name as it will appear at the
+#                             # dashboard (by default is the job_name)
+#                             # JOBs sharing a name are mutually exclusive
+#     update_every: 1         # the JOB's data collection frequency
+#     priority: 60000         # the JOB's order on the dashboard
+#     retries: 60             # the JOB's number of restoration attempts
+#     autodetection_retry: 0  # the JOB's re-check interval in seconds
+#
+# Additionally to the above, apache also supports the following:
```
Comment (l2isbad) [inline conf.d/python.d/powerdns_recursor.conf]: `apache also supports` :smile:

**Comment by l2isbad** [comment]
gj @lwhitworth :+1:

File: conf.d/python.d/powerdns_recursor.conf (hunk 1 above)
Comment (lwhitworth) [reply conf.d/python.d/powerdns_recursor.conf]: Copied the powerdns one that exists, same error carried forward (that'll teach me to not read it fully!!)

File: python.d/README.md [hunk 2]
```diff
@@ -2007,6 +2007,61 @@ local:
 
 ---
 
+# powerdns_recursor
+
+Module monitor powerdns recursor performance and health metrics.
+
+Following charts are drawn:
+
+1. **Questions In**
```
Comment (Ferroin) [inline python.d/README.md]: Is this whole 'Questions'/'Answers' thing the actual terms used by PowerDNS?  If so, it's fine to keep them, but if not, please either:

1. Update this to match up with the PowerDNS terminology.

or:

2. Use the actual DNS terminology ('Queries'/'Responses').

Comment (lwhitworth) [reply python.d/README.md]: https://doc.powerdns.com/recursor/metrics.html#gathered-information

Yep questions and answers are the terms used by the pdns-recursor API

**Comment by l2isbad** [comment]
> could be added as part of the existing PowerDNS plugin instead of needing to be a separate one

Nice idea! It's easy to do. I see we can distinguish between them by keys in result of `get_data()`. 

```python
ORDER_POWERDNS = [...]
CHARTS_POWERDNS = {...}

...


 def check(self):
    if condition():
        self.order = ORDER_POWERDNS 
        self.charts = CHARTS_POWERDNS
    else:
   ...
   
   return True
```

What doy you think @lwhitworth ?

**Comment by lwhitworth** [comment]
I'd keep them separate personally.  They are distinct solutions, you can run one or both on the same host so by keeping them separate you maintain/reinforce that distinction.  Additionally the API port is different between the two products (8081 for auth vs 8082 for recursor)

File: python.d/README.md (hunk 2 above)
Comment (Ferroin) [reply python.d/README.md]: Huh, that's a really odd choice of names.  Technically, it is semantically correct, it's just inconsistent with pretty much every other piece of DNS software in existence.

**Comment by l2isbad** [comment]
> you can run one or both on the same host

Every module support any number of jobs. One for port 8081, second for 8082. Module name can be changed in `check()` (powerdns=>powerdns_recursor)

**Comment by lwhitworth** [comment]
Fair enough then sounds like that covers it 👍

**Comment by l2isbad** [comment]
https://github.com/firehol/netdata/blob/master/python.d/apache.chart.py#L102

**Comment by Ferroin** [comment]
The one caveat is that this one may need to have some way to instantiate two copies if we want auto-detection for both the authoritative and recursive powerdns modules.

**Comment by l2isbad** [comment]
@lwhitworth do you need help with this?

**Comment by l2isbad** [comment]
1. Add powerdns recursos charts and order to the `powerdns.chart.py` (`POWERDNS_RECURSOS_CHARTS`, ...). 
2. Add a `check` method to the module.
3. Add powerdns recursor job to the module conf
4. Modify `README` in `python.d`

<details><summary>check method</summary>
<p>

```python
def check(self):
    d = self._get_data()
    if not d:
        return None
    if 'specific_to_powerdns_recursor_key` in d:  # you can add 2 check if you think it's needed
        self.order = POWERDNS_RECURSOR_ORDER
        self.charts = ...
        self.module_name = 'powerdns_recursor`
    return True
 ```

</p>
</details>

**Comment by l2isbad** [comment]
@lwhitworth i can do it myself if you want

**Pull request closed without merging**